### PR TITLE
Updating messaging for toHaveTagName to be more clear

### PR DESCRIPTION
**Pull request by GreenGremlin**
The messaging for `toHaveTagName` seems a bit confusing to me. From the current message, it's not very clear what didn't match. Looking at the enzyme API, `name()` will return `type.displayName -> type.name -> type`. I think display name summarizes all three of those well. This also makes me wonder if `toHaveTagName` should be renamed to `toHaveDisplayName` and `toHaveTagName` deprecated, since display name is what it is actually being checked.  Lastly, even though `===` was used for the comparison, it doesn't really offer to much insight as to why the match failed here, so I removed it.

#### Current messaging 

    Expected <Foo> node to equal (using ===) type "Bar" but it is a "Foo".
    Expected <Foo> node not to equal (using ===) type "Bar" but it is that type.

#### New messaging

    Expected <Foo> to have display name "Bar" but it had display name "Foo".
    Expected <Foo> not to have display name "Bar" but it did".

### CODE REVIEW

**Comment by blainekasten** [comment]
That's a decent argument. My next release will be a major. So it's probably fine to make the matcher name change.

In my mind, `toHaveDisplayName` is _more_ correct. But since enzyme is a jquery like API, it _feels_ like you are searching for a tag name. So it's not 100% clear cut to me. But probably better to be more correct and help educate people about React. So I'd be happy to accept a change of matcher name.

Want to do this in this PR?

**Comment by GreenGremlin** [comment]
Done. I also added a deprecate utility, which I'm using to print a deprecation warning when `toHaveTagName` is called. Ideally, the deprecation message would print after the tests are complete, but to accomplish that I think the deprecation functionality would have to be implemented in each environment. Let me know if you'd like this done another way, or if you have other ideas for implementing deprecation messaging.

**Comment by GreenGremlin** [comment]
> Maybe toHaveTagName could be preserved but refined to "really" check for tagName?
wrapper.getDOMNode().tagName

For clarity, are you saying it should only accept an element who's type is a string? Ultimately, I wouldn't mind that, but that change should probably not happen until after the deprecation has had time to propagate.

**Comment by blainekasten** [comment]
It looks like this has some changes that got copied in from your last PR. Can you rebase against master and squash down to 1 commit?

**Comment by blainekasten** [comment]
@pascalduez I don't see the benefits outweighing the risks of having 2 methods that are nearly named, and sometimes similar in functionality. I think it would just add some confusion. I think the display name works because it matches what's happening in React. And educating people more about React itself is helpful imo.

**Comment by GreenGremlin** [comment]
Done

File: packages/enzyme-matchers/src/assertions/toHaveTagName.js [hunk 1]
```diff
@@ -1,31 +1,15 @@
...
+
+const toHaveTagName = deprecate(
+  (enzymeWrapper: EnzymeObject, tag: string): Matcher => {
+    return toHaveDisplayName(enzymeWrapper, tag);
+  },
+  `Matcher ${colors.red(
+    'toHaveTagName'
+  )} is deprecated and will be removed in a future release. ` +
+    `Use the replacement, ${colors.blue('toHaveDisplayName')} instead.`
+);
```
Comment (blainekasten) [inline packages/enzyme-matchers/src/assertions/toHaveTagName.js]: love it. Nice work!

**Pull request merged**